src/main.cpp

```diff
@@ -76,28 +77,39 @@ Options: -h --help    Show this screen.
                 // version string, acquired from config.hpp via CMake
                 Yes::cmake::project_version)) };
 
-        std::array<double, 3> ring_offsets{ 0.0, 0.0, 0.0 };
+        std::array<double, 3> ring_offsets{ 0, 0, 0 };
         std::size_t current_ring{ 0 };
         auto screen{ ftxui::ScreenInteractive::TerminalOutput() };
 
+        // randomise the correct pin position
+        std::random_device random_device;
+        std::mt19937 random_engine(random_device());
+        std::uniform_int_distribution<int> endpoint_rng(1, 6);// NOLINT
+        std::array<int, 3> endpoints{
+            endpoint_rng(random_engine), endpoint_rng(random_engine), endpoint_rng(random_engine)
+        };
+
         const auto cylinder{ ftxui::Renderer(make_lock_cylinder(ring_offsets)) };
-        const auto graph{ ftxui::Renderer(make_plot_graph(current_ring, ring_offsets)) };
+        const auto graph{ ftxui::Renderer(make_plot_graph(current_ring, ring_offsets, endpoints)) };
         const auto text_box{ ftxui::Renderer(make_info_box(current_ring)) };
 
         const auto game_logic{ ftxui::Renderer([&] {
             if (current_ring >= ring_offsets.size()) {
                 screen.ExitLoopClosure()();
-                return ftxui::hbox(ftxui::text("GAME OVER")) | ftxui::border;
+                if (check_victory(ring_offsets, endpoints)) {
+                    return ftxui::hbox(ftxui::text("YOU WIN") | ftxui::borderDouble);
+                }
+                return ftxui::hbox(ftxui::text("GAME OVER")) | ftxui::borderDouble;
             }
-            return ftxui::hbox(cylinder->Render() | ftxui::borderDouble,
+            return ftxui::hbox(cylinder->Render(),
```

Comment:
all the double borders seem way too much to me, seems a lot cleaner to me with this removed, can mess with it more later on.